src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnectionOptions.cs

```diff
@@ -20,48 +21,110 @@ internal QuicConnectionOptions()
     /// Limit on the number of bidirectional streams the remote peer connection can create on an open connection.
     /// Default 0 for client and 100 for server connection.
     /// </summary>
-    public int MaxBidirectionalStreams { get; set; }
+    public int MaxInboundBidirectionalStreams { get; set; }
 
     /// <summary>
     /// Limit on the number of unidirectional streams the remote peer connection can create on an open connection.
     /// Default 0 for client and 10 for server connection.
     /// </summary>
-    public int MaxUnidirectionalStreams { get; set; }
+    public int MaxInboundUnidirectionalStreams { get; set; }
 
     /// <summary>
     /// Idle timeout for connections, after which the connection will be closed.
     /// Default <see cref="TimeSpan.Zero"/> means underlying implementation default idle timeout.
     /// </summary>
     public TimeSpan IdleTimeout { get; set; } = TimeSpan.Zero;
+
+    /// <summary>
+    /// Error code used when the stream needs to abort read or write side of the stream internally.
+    /// </summary>
+    // QUIC doesn't allow negative value: https://www.rfc-editor.org/rfc/rfc9000.html#integer-encoding
+    // We can safely use this to distinguish if user provided value during validation.
+    public long DefaultStreamErrorCode { get; set; } = -1;
+
+    /// <summary>
+    /// Error code used for <see cref="QuicConnection.CloseAsync(long, Threading.CancellationToken)"/> when the connection gets disposed.
+    /// To use different close error code, call  <see cref="QuicConnection.CloseAsync(long, Threading.CancellationToken)"/> explicitly before disposing.
+    /// </summary>
+    // QUIC doesn't allow negative value: https://www.rfc-editor.org/rfc/rfc9000.html#integer-encoding
+    // We can safely use this to distinguish if user provided value during validation.
+    public long DefaultCloseErrorCode { get; set; } = -1;
+
+    /// <summary>
+    /// Validates the options and potentially sets platform specific defaults.
+    /// </summary>
+    /// <param name="argumentName">Name of the from the caller.</param>
+    internal virtual void Validate(string argumentName)
+    {
+        if (MaxInboundUnidirectionalStreams < 0 || MaxInboundBidirectionalStreams > ushort.MaxValue)
+        {
+            throw new ArgumentOutOfRangeException($"{nameof(QuicConnectionOptions.MaxInboundBidirectionalStreams)} should be within [0, {ushort.Max}) range.", argumentName);
```

Comment:
Is this limitation of MsQuic? I think the RFC does not impose any limits on concurrently opened streams.

Reply:
```
uint16_t PeerBidiStreamCount;
uint16_t PeerUnidiStreamCount;
```
https://github.com/microsoft/msquic/blob/main/docs/api/QUIC_SETTINGS.md